dap_api/app/app/backend/geoutil.py

```diff
@@ -110,3 +113,45 @@ def meters_travelled(seconds, speed):
     @return:
     """
     return round(seconds * speed / 3600 * 1000)
+
+
+def build_diff_query(avoid_item, item, ors_api, ors_res_type) -> Function:
+    """
+    build the sql query to calculate the geometric difference between
+    corresponding isochrone or route items.
+    The returned query can be executed using a db session.
+    @param avoid_item: item of the ors response with avoid areas
+    @param item: item of the ors response without avoid areas
+    @param ors_api: ors service
+    @param ors_res_type: response format
+    @return: sqlalchemy query function
+    """
+    geom_no_avoid = get_geom_from_item(item, ors_res_type)
+    geom = get_geom_from_item(avoid_item, ors_res_type)
+    difference = None
+    if ors_api == "isochrones":
+        difference = func.ST_Difference(geom_no_avoid, geom)
+    elif ors_api == "directions":
+        difference = func.ST_Difference(geom, geom_no_avoid)
```

Comment:
This is confusing enough that I'd recommend a comment explaining why it has to be the other way round for directions.

Reply:
agreed